### ISSUE TITLE
`nativeEnum()` value with description

### ISSUE THREAD

**Issue by j-murata** [sachinraja/zod-to-ts#30]
It seems that adding a description to the `nativeEnum()` value field outputs `unknown` as shown below.
Is this a bug or expected behavior?

<p><details open>
<summary><code>enum-value.mjs</code></summary>

```js
import { z } from 'zod';
import { printNode, withGetType, zodToTs } from 'zod-to-ts';

const Enum = z.nativeEnum({
    ONE: 1,
    TWO: 2,
}).describe('Comment for Enum');

withGetType(Enum, ts => ts.factory.createIdentifier('Enum'));

const schema = z.object({
    key: Enum,
});

const { node } = zodToTs(schema, { resolveNativeEnums: true });
console.log(printNode(node));
// {
//     /** Comment for Enum */
//     key: Enum;
// }
```

</details></p>

<p><details open>
<summary><code>enum-value-with-description.mjs</code></summary>

```js
import { z } from 'zod';
import { printNode, withGetType, zodToTs } from 'zod-to-ts';

const Enum = z.nativeEnum({
    ONE: 1,
    TWO: 2,
});

withGetType(Enum, ts => ts.factory.createIdentifier('Enum'));

const schema = z.object({
    key: Enum.describe('Comment for key'),
});

const { node } = zodToTs(schema, { resolveNativeEnums: true });
console.log(printNode(node));
// {
//     /** Comment for key */
//     key: unknown;
// }
```

</details></p>

**Comment by j-murata** [comment]
I looked into it in depth.
It seems to be due to the fact that different instances are returned by `.describe()`.

```js
const Enum1 = z.nativeEnum({
    ONE: 1,
    TWO: 2,
});
const Enum2 = Enum1.describe('Comment for Enum');
console.log(Enum1 === Enum2); // false
```

[Currently `withGetType()` adds `getType` directly to the schema object](https://github.com/sachinraja/zod-to-ts/blob/8fe9eaa4cd2073b2ef412ade0f403fa765dd7fc2/src/utils.ts#L40-L43), but I think this problem could probably be solved if it were managed by the `schema._def` object instead.

Until then, the only workaround seems to be to override `.describe()` as follows:
```js
{
    const originalDescribe = ZodType.prototype.describe;
    console.log('override `ZodType.prototype.describe()`');
    ZodType.prototype.describe = function (...args) {
        const returnValue = originalDescribe.call(this, ...args);
        returnValue.getType = this.getType;
        return returnValue;
    }
}
```

**Comment by sachinraja** [comment]
thanks for investigating, I implemented your fix in 0fee6095405ef8e06ce70500fd6d1317b2de1e17
